### PR TITLE
Receive flow control on ack

### PR DESCRIPTION
**Pull request by noodlefrenzy**
Fix for #210

### CODE REVIEW

File: lib/link.js [hunk 1]
```diff
@@ -146,7 +146,7 @@ Link.prototype._attachReceived = function(attachFrame) {
   this.emit(Link.Attached, this);
   this._resolveAttachPromises(null, this);
 
-  this._checkCredit();
+  this._checkCredit(/* initial = */ true);
```
Comment (mbroadst) [inline lib/link.js]: is there any detriment to doing: `this._checkCredit({ initial: true });` here? for the pure sake of readability.

Comment (noodlefrenzy) [inline lib/link.js]: I don't think so other than object instantiation vs. boolean, but since it's only on attach it's not like this is inner-loop code.

**Comment by mbroadst** [comment]
that's my only real comment, this is pretty straightforward. LGTM

**Comment by mbroadst** [comment]
Other than it might be nice to have a test for it? That might be a bit involved though

**Pull request merged**